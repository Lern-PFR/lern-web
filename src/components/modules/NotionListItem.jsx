import PropTypes from 'prop-types';
import { StyledList, StyledListItem } from 'components/shared/styledElements';
import { withTranslation } from 'react-i18next';
import routes from 'routes/keys';

import { BodyCopy, Paragon } from 'components/shared/typography';
import { OutlinedButton } from 'components/shared/buttons';
import { Link } from 'components/shared/navigation';

import { lessonList, notionDataContainer, notionDescription, notionName, startNotionButton } from 'theme/pages/modules/moduleDetailsPage';

import LessonListItem from './LessonListItem';

/**
 * @name NotionListItem
 * @description An item used to display information about a notion and its composing lessons.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {string|number}	id			The current notion's id.
 * @param {string}			name		The current notion's name.
 * @param {string}			description	The current notion's description.
 * @param {array}			[lessons]	The current notion's lessons.
 * @param {func}			t			The translation method provided by the withTranslation HoC.
 */
const NotionListItem = ({ id, name, description, lessons, t }) => (
	<StyledListItem {...notionDataContainer}>
		<Paragon tag="h2" {...notionName}>{name}</Paragon>
		<BodyCopy {...notionDescription}>{description}</BodyCopy>
		<OutlinedButton {...startNotionButton}>
			<Link to={routes.notions.notionDetails.replace(':notionId', id)}>
				{t('notions.links.start')}
			</Link>
		</OutlinedButton>
		<StyledList listStyle="none" {...lessonList}>
			{lessons && lessons.map(({ id: lessonId, name: lessonName, description: lessonDesc }) => (
				<LessonListItem key={lessonId} id={lessonId} name={lessonName} notionId={id} description={lessonDesc} />
			))}
		</StyledList>
	</StyledListItem>
);
NotionListItem.propTypes = {
	id: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
	]).isRequired,
	name: PropTypes.string.isRequired,
	description: PropTypes.string.isRequired,
	lessons: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired,
		description: PropTypes.string.isRequired,
	})),
	t: PropTypes.func.isRequired,
};

NotionListItem.defaultProps = {
	lessons: [],
};

export default withTranslation()(NotionListItem);
