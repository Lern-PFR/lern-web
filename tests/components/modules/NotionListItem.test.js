import { shallow } from 'enzyme';
import NotionListItem from 'components/modules/NotionListItem';

describe('NotionListItem', () => {
	it('should match previous snapshot', () => {
		const sut = (<NotionListItem id="abcd" name="abcd" description="dummy description" />);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot with lessons', () => {
		const lessons = [
			{ id: '1', name: 'lesson 1', description: 'Lorem ipsum dolor sit amet.' },
			{ id: '2', name: 'lesson 2', description: 'Lorem ipsum dolor sit amet.' },
			{ id: '3', name: 'lesson 3', description: 'Lorem ipsum dolor sit amet.' },
		];

		const sut = (<NotionListItem id="abcd" name="abcd" description="dummy description" lessons={lessons} />);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});
});
