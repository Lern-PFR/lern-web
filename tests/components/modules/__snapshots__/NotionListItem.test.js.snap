// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConceptListItem should match previous snapshot 1`] = `
<styled.li
  borderTop="solid 1px #DBD9D2"
  paddingBottom="3em"
  paddingTop="2em"
>
  <Paragon
    lineHeight="initial"
    margin={0}
    tag="h2"
    textTransform="capitalize"
  />
  <BodyCopy
    lineHeight="initial"
    margin={0}
  >
    dummy description
  </BodyCopy>
  <OutlinedButton
    marginTop="1em"
  >
    <Link
      to="/concepts/abcd"
    >
      concepts.links.start
    </Link>
  </OutlinedButton>
  <styled.ul
    display="grid"
    gridAutoFlow="row"
    gridGap="1em"
    gridTemplateColumns="repeat(auto-fill, minmax(250px, 1fr))"
    listStyle="none"
    marginTop="2em"
    padding={0}
  />
</styled.li>
`;

exports[`ConceptListItem should match previous snapshot with lessons 1`] = `
<styled.li
  borderTop="solid 1px #DBD9D2"
  paddingBottom="3em"
  paddingTop="2em"
>
  <Paragon
    lineHeight="initial"
    margin={0}
    tag="h2"
    textTransform="capitalize"
  />
  <BodyCopy
    lineHeight="initial"
    margin={0}
  >
    dummy description
  </BodyCopy>
  <OutlinedButton
    marginTop="1em"
  >
    <Link
      to="/concepts/abcd"
    >
      concepts.links.start
    </Link>
  </OutlinedButton>
  <styled.ul
    display="grid"
    gridAutoFlow="row"
    gridGap="1em"
    gridTemplateColumns="repeat(auto-fill, minmax(250px, 1fr))"
    listStyle="none"
    marginTop="2em"
    padding={0}
  >
    <LessonListItem
      conceptId="abcd"
      description="Lorem ipsum dolor sit amet."
      disabled={false}
      id="1"
      key="1"
      name="lesson 1"
      t={[Function]}
    />
    <LessonListItem
      conceptId="abcd"
      description="Lorem ipsum dolor sit amet."
      disabled={false}
      id="2"
      key="2"
      name="lesson 2"
      t={[Function]}
    />
    <LessonListItem
      conceptId="abcd"
      description="Lorem ipsum dolor sit amet."
      disabled={false}
      id="3"
      key="3"
      name="lesson 3"
      t={[Function]}
    />
  </styled.ul>
</styled.li>
`;
